### PR TITLE
bpo-26120: make pydoc exclude __future__ imports from the data block of the module

### PR DESCRIPTION
**Pull request by iritkatriel**
https://bugs.python.org/issue26120

### CODE REVIEW

**Comment by iritkatriel** [comment]
> Python 2.7 [reached end-of-life January 1, 2020](https://devguide.python.org/devcycle/#end-of-life-branches). Do we really need to carry `__future__` around in pydoc?

The __future__ module is not a 2.7 feature, it's still used (like ``__annotations__``)

File: Lib/pydoc.py [hunk 1]
```diff
@@ -284,6 +285,9 @@ def visiblename(name, all=None, obj=None):
     # Namedtuples have public fields and methods with a single leading underscore
     if name.startswith('_') and hasattr(obj, '_fields'):
         return True
+    if name in __future__.all_feature_names:
```
Comment (Fidget-Spinner) [inline Lib/pydoc.py]: At the risk of sounding like a pedant, would making this a set outside the function and checking it here be faster (however marginally)?

**Review by Fidget-Spinner — COMMENTED** [review]
This LGTM, but since I never touch pydoc apart from the occasional security fix, I'm refraining from approving.

Also, do we need tests for this? Or does pydoc not test this function at all?

> Python 2.7 reached end-of-life January 1, 2020. Do we really need to carry __future__ around in pydoc?

Of course! I still want my `from __future__ import braces` and `from __future__ import barry_as_FLUFL` ;-).

That aside, yeah Irit added `print_function` to test that the code works as intended, but it could be any recent non-2.7 `__future__`.

File: Lib/pydoc.py (hunk 1 above)
Comment (iritkatriel) [reply Lib/pydoc.py]: > do we need tests for this?

It is tested - if you take the change I made to Lib/test/pydoc_mod.py to the main branch the test will fail.

> At the risk of sounding like a pedant, would making this a set outside the function and checking it here be faster (however marginally)?

Sure, there's nothing to lose.

File: Lib/pydoc.py [hunk 2]
```diff
@@ -288,6 +291,9 @@ def visiblename(name, all=None, obj=None):
     # Namedtuples have public fields and methods with a single leading underscore
     if name.startswith('_') and hasattr(obj, '_fields'):
         return True
+    if name in _future_feature_names:
```
Comment (Fidget-Spinner) [inline Lib/pydoc.py:295]: Perhaps a comment would help. This looks confusing on its own.
```suggestion
    # Ignore __future__ imports.
    if name in _future_feature_names:
```

**Review by Fidget-Spinner — APPROVED** [review]
I don't think anyone is actively maintaining pydoc. LGTM on my end.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**